CONTRIBUTING.rst

```diff
@@ -36,6 +39,47 @@ If modifying dependencies in ``mlflow/server/js/package.json``, run ``npm update
 ``mlflow/server/js`` to install the updated dependencies.
 
 
+Java API
+--------
+
+The MLflow Java client depends on the Python client, so first install the Python client in a conda
+environment as described above. `Install <https://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+the Java 8 JDK (or above), and `download <https://maven.apache.org/download.cgi>`_
+and `install <https://maven.apache.org/install.html>`_ Maven. You can then build and run tests via:
+
+.. code:: bash
+
+  cd mlflow/java
```

Comment:
This is not the client, btw -- the client is in `mlflow/java/client`. Maybe we can add a description under the Java section that "Certain MLflow modules are implemented in Java, under the `mlflow/java/` directory. These are the Java Tracking API client (mlflow/java/client) and the Model Scoring Server for Java-based models like MLeap (mlflow/java/scoring).

We can keep the `cd mlflow/java` and just mention that it builds _all_ Java projects.